GitCommands/FullPathResolver.cs

```diff
@@ -56,8 +56,8 @@ public string Resolve(string path)
                 basePath += Path.DirectorySeparatorChar;
             }
 
-            var uri = new Uri(new Uri(basePath), path);
-            return uri.LocalPath;
+            var uri = Path.Combine(basePath, path);
+            return uri;
```

Comment:
We must be absolutely certain we aren't regressing (e.g. #6702).

Reply:
#6702 is a part of the problem as it uses Uri() to get the path and Uri().LocalPath has different rules than what the file system allows.

Uri() is used also in Clone, where there has been a number of illegal char handling recently, why this is maybe should be changed there too.